device/transport/amqp/src/amqp.ts

```diff
@@ -67,7 +70,59 @@ export class Amqp extends EventEmitter implements Client.Transport, StableConnec
     });
 
     this._deviceMethodClient = new AmqpDeviceMethodClient(this._config, this._amqp);
-    this._receiver = new AmqpReceiver(this._config, this._amqp, this._deviceMethodClient);
+    this._deviceMethodClient.on('error', (err) => {
+      this.emit('errorReceived', err);
+    });
+
+    this._c2dEndpoint = endpoint.messagePath(encodeURIComponent(this._config.deviceId));
+    this._d2cEndpoint = endpoint.eventPath(this._config.deviceId);
+
+    const c2dErrorListener = (err) => {
+      debug('Error on the C2D link: ' + err.toString());
+      // this is the right thing to do (even better would be to emit 'error' directly but can't be done right now because the client does not have the proper error handling logic.)
+      // this will come with the retry logic.
+      // this.emit('errorReceived', err);
+    };
+
+    const c2dMessageListener = (msg) => {
+      this.emit('message', msg);
+    };
+
+    /*Codes_SRS_NODE_DEVICE_AMQP_RECEIVER_16_008: [The `Amqp` object shall remove the listeners on `message` and `error` events of the underlying `ReceiverLink` when no-one is listening to its own `message` event.]*/
+    this.on('removeListener', (eventName, eventCallback) => {
+      if (eventName === 'message') {
+        if (this._c2dLink) {
```

Comment:
>this._c2dLink) [](start = 12, length = 14)

Do you need to null this on the remove?

Reply:
probably a good idea...

---
In reply to: [140297588](https://github.com/Azure/azure-iot-sdk-node/pull/87#discussion_r140297588) [](ancestors = 140297588)